File: DSA/AnshulJain_590014934/Day_43_Question_3.py
from collections import deque
queue = deque(input("Enter characters without spaces: "))#Taing input from the user
stack = []
for ch in queue:
    stack.append(ch)
is_palindrome = True
for ch in queue:
    if ch != stack.pop():
        is_palindrome = False
        break
print("Yes" if is_palindrome else "No")

File: DSA/AdityaSingh_590014769/Day_43_Question_3.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

bool isPalindrome(char queue[], int n) {
    for (int i = 0; i < n / 2; i++) {
        if (queue[i] != queue[n - 1 - i]) {
            return false;
        }
    }
    return true;
}

int main() {
    char queue[] = {'r', 'a', 'c', 'e', 'c', 'a', 'r'};
    int n = sizeof(queue) / sizeof(queue[0]);

    if (isPalindrome(queue, n))
        printf("Yes\n");
    else
        printf("No\n");

    return 0;
}
